src/System.Private.CoreLib/shared/System/Text/EncodingProvider.cs

```diff
@@ -11,29 +12,29 @@ namespace System.Text
     public abstract class EncodingProvider
     {
         public EncodingProvider() { }
-        public abstract Encoding GetEncoding(string name);
-        public abstract Encoding GetEncoding(int codepage);
+        public abstract Encoding? GetEncoding(string name);
+        public abstract Encoding? GetEncoding(int codepage);
 
         // GetEncoding should return either valid encoding or null. shouldn't throw any exception except on null name
-        public virtual Encoding GetEncoding(string name, EncoderFallback encoderFallback, DecoderFallback decoderFallback)
+        public virtual Encoding? GetEncoding(string name, EncoderFallback encoderFallback, DecoderFallback decoderFallback)
         {
-            Encoding enc = GetEncoding(name);
+            Encoding? enc = GetEncoding(name);
             if (enc != null)
             {
-                enc = (Encoding)GetEncoding(name).Clone();
+                enc = (Encoding)GetEncoding(name)!.Clone(); // TODO-NULLABLE: this could be enc.Clone()
```

Comment:
If @tarekgh agrees, just go ahead and fix that here.

Reply:
I believe that will work. let's do it.

---
In reply to: [272199546](https://github.com/dotnet/coreclr/pull/23708#discussion_r272199546) [](ancestors = 272199546)